relay-server/src/metric_stats.rs

```diff
@@ -61,19 +62,23 @@ impl MetricStats {
     }
 
     /// Tracks the metric volume and outcome for the bucket.
-    pub fn track_metric(&self, scoping: Scoping, bucket: Bucket, outcome: Outcome) {
+    pub fn track_metric<'a, V>(&self, scoping: Scoping, bucket: V, outcome: Outcome)
+    where
+        V: Into<BucketView<'a>>,
+    {
         if !self.is_enabled(scoping) {
             return;
         }
 
-        let Some(volume) = self.to_volume_metric(&bucket, &outcome) else {
+        let bucket_view = bucket.into();
```

Comment:
I'd just shadow bucket here:

```suggestion
        let bucket = bucket.into();
```